File: tests/combi/variable_inline_and_rename.cpp


int test_const_const()
{
  const int x = 5;
  const int y = x;
  int z = x + y;
}

int test_nonconst_const()
{
  int x = 5;
  x = 6;
  x = 7;
  const int y = x;
  int z = x + y;
}

int test_const_nonconst()
{
  const int x = 5;
  int y = x;
  int z = x + y;
}

int test_nonconst_nonconst()
{
  int x = 5;
  x = 6;
  x = 7;
  int y = x;
  int z = x + y;
}

typedef struct {
  int x;
  int y;
} vect;

int test_nonconstvect_nonconstvect(){
  vect a;
  a.x = 6;
  a.x = 7;
  vect b = a;
  int c = b.x;
  int d = b.y;
}

File: tests/combi/variable_inline_and_rename.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun () -> 

  !! Variable.inline_and_rename [cVarDef "y"];

)

"
int main(){
  const int x = 5;
  const int y = x;
  int z = x + y;
}
"

let _ = Run.script_cpp (fun _ ->

  !! Variable.inline_and_rename [cTopFunDef "test_const_const"; cVarDef "y"];
  !! Variable.inline_and_rename [cTopFunDef "test_nonconst_const"; cVarDef "y"];
  !! Variable.inline_and_rename [cTopFunDef "test_const_nonconst"; cVarDef "y"];
  !! Variable.inline_and_rename [cTopFunDef "test_nonconst_nonconst"; cVarDef "y"];

  !! Variable.inline_and_rename [cTopFunDef "test_const_const_vect"; cVarDef "b"];
  !! Variable.inline_and_rename [cTopFunDef "test_nonconst_const_vect"; cVarDef "b"];
  !! Variable.inline_and_rename [cTopFunDef "test_const_nonconst_vect"; cVarDef "b"];
  !! Variable.inline_and_rename [cTopFunDef "test_nonconst_nonconst_vect"; cVarDef "b"];

)
